File: integration_example.js
#!/usr/bin/env node
/**
 * JavaScript/Node.js integration example for Bedrock Nova Proxy
 * npm install openai
 */

const OpenAI = require('openai');

// Your Bedrock Nova Proxy configuration
const client = new OpenAI({
    baseURL: 'https://itw9z9jxai.execute-api.eu-north-1.amazonaws.com/prod/v1',
    apiKey: 'dummy' // Not used but required
});

async function exampleChat() {
    console.log('🤖 Regular Chat Completion:');
    
    try {
        const response = await client.chat.completions.create({
            model: 'gpt-4o-mini', // Maps to eu.amazon.nova-lite-v1:0
            messages: [
                { role: 'system', content: 'You are a helpful assistant.' },
                { role: 'user', content: 'Explain serverless computing in simple terms.' }
            ],
            max_tokens: 200,
            temperature: 0.7
        });
        
        console.log('Response:', response.choices[0].message.content);
        console.log('Tokens used:', response.usage);
        
    } catch (error) {
        console.error('Error:', error.message);
    }
}

async function exampleStreaming() {
    console.log('\n🌊 Streaming Response:');
    
    try {
        const stream = await client.chat.completions.create({
            model: 'gpt-4o-mini',
            messages: [
                { role: 'user', content: 'Count from 1 to 5 with explanations.' }
            ],
            stream: true,
            max_tokens: 150
        });
        
        for await (const chunk of stream) {
            if (chunk.choices[0]?.delta?.content) {
                process.stdout.write(chunk.choices[0].delta.content);
            }
        }
        console.log('\n');
        
    } catch (error) {
        console.error('Streaming error:', error.message);
    }
}

async function main() {
    console.log('🚀 Bedrock Nova Proxy JavaScript Integration Examples');
    console.log('='.repeat(60));
    
    await exampleChat();
    await exampleStreaming();
    
    console.log('✅ Integration examples completed!');
}

if (require.main === module) {
    main().catch(console.error);
}

File: integration_example.py
#!/usr/bin/env python3
"""
Integration example for Bedrock Nova Proxy
Replace your OpenAI calls with this configuration
"""

from openai import OpenAI

# Your Bedrock Nova Proxy configuration
client = OpenAI(
    base_url="https://itw9z9jxai.execute-api.eu-north-1.amazonaws.com/prod/v1",
    api_key="dummy"  # Not used but required by OpenAI client
)

def example_chat():
    """Example chat completion"""
    response = client.chat.completions.create(
        model="gpt-4o-mini",  # Maps to eu.amazon.nova-lite-v1:0
        messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": "What are the benefits of using AWS Bedrock?"}
        ],
        max_tokens=200,
        temperature=0.7
    )
    
    print("Response:", response.choices[0].message.content)
    print("Tokens used:", response.usage)

def example_streaming():
    """Example streaming response"""
    print("Streaming response:")
    
    stream = client.chat.completions.create(
        model="gpt-4o-mini",
        messages=[
            {"role": "user", "content": "Write a short poem about cloud computing"}
        ],
        stream=True,
        max_tokens=150
    )
    
    for chunk in stream:
        if chunk.choices[0].delta.content is not None:
            print(chunk.choices[0].delta.content, end="", flush=True)
    print("\n")

if __name__ == "__main__":
    print("🚀 Bedrock Nova Proxy Integration Examples")
    print("=" * 50)
    
    print("\n1. Regular Chat Completion:")
    example_chat()
    
    print("\n2. Streaming Response:")
    example_streaming()
    
    print("\n✅ Integration examples completed!")
